app/equipment/armor.py

```diff
@@ -4,21 +4,15 @@
 
 
 class Armor(Equipment):
-    def __init__(self, part: str, power: int) -> None:
+    def __init__(self, part: str, protection: int) -> None:
         super().__init__(part)
-        self.power = power
+        self.power = protection
 
     @classmethod
     def apply(cls, knight: Any, source: dict) -> None:
-        armor_data = (
-            source.get(knight.name.lower(),
-                       source.get(knight.name.replace(" ", "_").lower(), {})
-                       ).get("armour", [])
-        )
-
-        if armor_data:
+        if source.get("armour"):
             knight.equip_list = []
-            for armor in armor_data:
-                armor = Armor(part=armor["part"], power=armor["protection"])
+            for armor in source.get("armour"):
+                armor = Armor(armor["part"], armor["protection"])

```

Comment:
Invite you to attend today's QnA at 14:00, will show you some trick ;)
Or you can use this tip: you can use unpacking here to make it look better